### PR TITLE
Define RP processing of be and bs flags during `.create()` and `.get()`

### PR DESCRIPTION
**Pull request by MasterKale**
The spec was light on details on how to handle a couple of states of the `be` and `bs` flags coming out of calls to `.create()` and `.get()`. This PR tries to clarify RP handling of potential combinations of these flags during registration and authentication:

1. A device-bound credential indicates that it is backed up during registration (`be:0+bs:1`)
2. A device-bound credential indicates that it is backed up after authentication (`be:0+bs:1`)
3. A credential's backup eligibility changes after registration (`be:0 -> be:1` or `be:1 -> be:0`)

This should address #1791.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/pull/1907.html" title="Last updated on Jul 12, 2023, 1:54 PM UTC (4514f5f)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/1907/6dfbdba...4514f5f.html" title="Last updated on Jul 12, 2023, 1:54 PM UTC (4514f5f)">Diff</a>

### CODE REVIEW

**Review by Firstyear — APPROVED** [review]
This addresses my concerns yes :)

**Review by timcappalli — APPROVED** [review]

File: index.bs
Comment (emlun) [inline index.bs:5534]: The new step during registration is always in force, but these criteria are only enforced "if the credential backup state is used as part of Relying Party business logic or policy". We should have the same enforcement criteria in both cases: either always, or only "if the credential backup state is used as part of Relying Party business logic or policy".

I would favour always rejecting `BE=0; BS=1`, but the comparison with `|credentialRecord|.[$credential record/backupEligible$]` probably needs to stay within the qualified case since it's optional for the RP to store that value.

**Review by emlun — CHANGES_REQUESTED** [review]

File: index.bs
Comment (MasterKale) [reply index.bs]: I went ahead and made this change

File: index.bs
Comment (MasterKale) [reply index.bs:5536]: I tried to address this in https://github.com/w3c/webauthn/pull/1907/commits/3e0395f5d5cdb6cac5b814ef9492be2f803f8897, basically copy-pasted your suggested wording in the comment above to above these conditional checks based on `credentialRecord`. That should enforce the "impossibility" of an authenticator returning `be:0,bs:1` too.

Comment (emlun) [reply index.bs:5536]: Looks good! The third conditional check can be eliminated now, see new thread.

File: index.bs
Comment (emlun) [inline index.bs:5534]: This can be eliminated now.

```suggestion
```

Reasoning:

 1. Let `be=0 => bs=0` denote the requirement `1. If the [=authData/flags/BE=] bit of the <code>[=flags=]</code> in |authData| is not set, verify that the [=authData/flags/BS=] bit is not set.`.
 2. Let `BE=1 => be=1` denote the requirement `1. If <code>|credentialRecord|.[$credential record/backupEligible$]</code> is set, verify that |currentBe| is set.`.
 3. Let `BE=0 => be=0` denote the requirement `1. If <code>|credentialRecord|.[$credential record/backupEligible$]</code> is not set, verify that |currentBe| is not set.`.
 4. Let `BE=0 => bs=0` denote the requirement `1. If <code>|credentialRecord|.[$credential record/backupEligible$]</code> is not set, verify that |currentBs| is not set.`
 5. Collect equations:
    ```
    be=0 => bs=0
    BE=1 => be=1
    BE=0 => be=0
    BE=0 => bs=0
    ```
    Rearrange:
    ```
    BE=1 => be=1

    BE=0 => be=0
    be=0 => bs=0

    BE=0 => bs=0
    ```
    We see that `BE=0` implies `bs=0` through two paths, i.e. the fourth equation is redundant.

**Review by emlun — CHANGES_REQUESTED** [review]

File: index.bs
Comment (MasterKale) [reply index.bs:5534]: Ah, I see what you mean, the authentication response would fail verification earlier when we assert that `be=0` and `bs=1` is bad, before there was a chance for `BE=0` and `bs=1` to assert a failure.

Comment (MasterKale) [reply index.bs]: I removed the redundant step here: https://github.com/w3c/webauthn/pull/1907/commits/4514f5fcefa2c8d8e23fbd8e8b1cbc3990ce959d

**Comment by MasterKale** [comment]
Confirmed during the meeting that since no objections came up on 7/21 it's okay to merge this outside of a WAWG meeting, once @emlun approves.

**Review by emlun — APPROVED** [review]
LGTM, thanks!

**Review by ve7jtb — APPROVED** [review]

**Pull request merged**
